minisom.py

```diff
@@ -390,15 +391,23 @@ def activation_response(self, data):
         for x in data:
             a[self.winner(x)] += 1
         return a
+    
+    def dist_mat(self, data):
+        """
+        calculate distance matrix: dist[i,j]: i: the i-th data, j: the j-th node
+        """
+        D = np.array(data)
+        C = self._weights.reshape(-1, self._weights.shape[2])
```

Comment:
let's use explicit names, for example
`weights_flat = self._weights.reshape(-1, self._weights.shape[2])`